File: Dataset/Leetcode/test/101/610.py
 class Solution:
    def XXX(self, root: TreeNode) -> bool:
        x1 = self.zx(root)
        x2 = self.jx(root)
        return x1 == x2
    def zx(self, root):
        if not root:
            return ['null']
        return self.zx(root.left) + [root.val] + self.zx(root.right)
    def jx(self, root):
        if not root:
            return ['null']
        return self.jx(root.right) + [root.val] + self.jx(root.left)

File: Dataset/Leetcode/train/110/610.cpp
 class Solution {
public:
    int get_height(TreeNode* node){
        if(!node)
            return 0;
        
        int left = get_height(node->left);
        int right = get_height(node->right);

        if(abs(left - right) > 1 || left == -1 || right == -1)
            return -1;
        
        return max(left, right) + 1;
    }

    bool XXX(TreeNode* root) {
        return get_height(root) != -1;
    }
};

